CMakeLists.txt

```diff
@@ -188,12 +194,34 @@ endif()
 set(Kokkos_CXX_STANDARD 14)
 if(EXISTS ${Kokkos_ROOT}/CMakeLists.txt)
   add_subdirectory(${Kokkos_ROOT} Kokkos)
+  
+  #  get_target_property(Kokkos_SOURCES Kokkoscore SOURCES)
 elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt)
   add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos)
+  #add_library(lkokkoscore STATIC IMPORTED)
+  #  set_property(TARGET lkokkoscore PROPERTY IMPORTED_LOCATION_RELEASE ${CMAKE_BINARY_DIR}/Kokkos/core/src/libkokkoscore.a)
+
+  #add_library(kokkoscore STATIC IMPORTED)
+  #  get_target_property(Kokkos_SOURCES Kokkoscore SOURCES)
 else()
   find_package(Kokkos 3 REQUIRED)
 endif()
 
+#add_library(Kokkos SHARED IMPORTED)
+#add_library(Kokkos INTERFACE)
+#target_link_libraries(Kokkos INTERFACE ${Kokkos_LIBRARIES})
+#target_compile_definitions(Kokkos INTERFACE ${Kokkos_DEFINITIONS})
+#target_include_directories(Kokkos INTERFACE ${Kokkos_INCLUDE_DIRS})
+#target_sources(Kokkos INTERFACE ${Kokkos_SOURCES})
```

Comment:
@brryan you can also treat Kokkos as interface library similar to HDF5_C, I just don't know the consequences of doing this, I think I also got your sample working doing this. You need to add these lines and the install command below I think.